### ISSUE TITLE
ClientInfo not received in the response from AAD

### ISSUE THREAD

**Issue by kevin-mcmanus** [AzureAD/microsoft-authentication-library-for-js#1703]
I'm trying to obtain an access token using `.getAccessToken(args)` from the `react-aad-msal` npm package.  The console is reporting a couple of errors that originate from this repo:

`[ERROR] ClientAuthError: The service did not populate client_info in the response, Please verify with the service team Failed with error: ClientInfo not received in the response from the server`

And

`VM443 1.chunk.js:27570 Uncaught (in promise) ClientAuthError: The service did not populate client_info in the response, Please verify with the service team Failed with error: ClientInfo not received in the response from the server
    at ClientAuthError.AuthError [as constructor] (http://localhost:3000/static/js/1.chunk.js:27570:24)
    at new ClientAuthError (http://localhost:3000/static/js/1.chunk.js:27715:24)
    at Function.ClientAuthError.createClientInfoNotPopulatedError (http://localhost:3000/static/js/1.chunk.js:27798:12)
    at MsalAuthProvider.UserAgentApplication.saveTokenFromHash (http://localhost:3000/static/js/1.chunk.js:25782:93)
    at MsalAuthProvider.UserAgentApplication.processCallBack (http://localhost:3000/static/js/1.chunk.js:25245:23)
    at MsalAuthProvider.UserAgentApplication.handleAuthenticationResponse (http://localhost:3000/static/js/1.chunk.js:25299:10)
    at MsalAuthProvider.<anonymous> (http://localhost:3000/static/js/1.chunk.js:25034:20)
    at step (http://localhost:3000/static/js/1.chunk.js:84333:17)
    at Object.next (http://localhost:3000/static/js/1.chunk.js:84264:14)
    at fulfilled (http://localhost:3000/static/js/1.chunk.js:84217:24)`

Stepping through the source code:

https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/18c3f99fa1a3877c7965963de45db4e65a9d9c43/lib/msal-core/src/UserAgentApplication.ts#L1617

I can see that `hashParams` doesn't contain a `client_info`, which is why I get the error. 

However, the comment:
`// retrieve client_info - if it is not found, generate the uid and utid from idToken`
suggests that it will attempt to generate the client info from the `id_token` which I do have.  Unfortunately, from what I can tell, the code logs the error and throws an exception without trying to create the client info.

What is the intended behaviour?  Should the library attempt to create the object from the `id_token` (as per the comment), or does `client_info` really need to be set correctly up stream?

**Comment by jasonnutter** [comment]
@kevin-mcmanus Are you using ADFS?

Also, in the future please use of the issue templates, thanks!

**Comment by kevin-mcmanus** [comment]
@jasonnutter ,

No I'm not using ADFS, my scenario is more obscure than that...

In production I am using Azure Active Directory with no federation or connection to an on-prem directory.

For development I'm using a 100% Identity Server 4 (https://github.com/IdentityServer/IdentityServer4) solution, with users and resources all configured in IS4 with no links to AAD (or ADFS).

AzureAD is being used to protect a React SPA.  Ideally I want to use the same libraries to secure both prod and non-prod environments, with just a configuration change or two required (e.g. the authority URL).

This setup started out as a trial, with no expectations that it would even work.  I was simply going on the basis that both support OIDC & OAuth 2.  Initial development required a login to access the site, for which AzureAD performed well.  (The only non standard IS4 change was a very minor URL rewrite.)

I am now extending the use of AzureAD and am looking to secure access to a backend REST API and give it the ability to identify the current user.  This is where I've hit the issue described above (in my dev environment).

Looking at the code, I wonder if I've hit an implementation bug rather than a product limitation.  It would be great to get some clarity to help decide on my next steps.

N.B. I acknowledge this is not a supported use case and can take a look at alternatives if necessary.

Since submitting my request I've now found the issue templates that you refer to; I'd be happy to use these in the future.  Please be aware that they aren't presented as an option when viewing the code, selecting a line, clicking '...' and selecting 'Reference in a new issue'.

**Comment by jasonnutter** [comment]
@kevin-mcmanus Thanks for the context. It looks like the comment in question should have been removed/updated when the exception was added, so we'll make sure that gets fixed. 

Currently, the library is expecting responses to include `client_info`, but #1668 will remove that requirement in order to properly support ADFS, so that may unblock your scenario as well.

**Comment by kevin-mcmanus** [comment]
@jasonnutter Thanks for the reply; that sounds very promising  I'll retest when the referenced ADFS code is released.

**Comment by jmckennon** [comment]
This issue may be solved by #1668, but IdentityServer is not officially supported by Microsoft. Please open a new issue if you have further questions, thanks!
